src/nbl/video/CVKLogicalDevice.h

```diff
@@ -178,52 +802,382 @@ class CVKLogicalDevice final : public ILogicalDevice
+        }
+        else
+        {
+            return nullptr;
+        }
     }
 
     core::smart_refctd_ptr<IGPUImageView> createGPUImageView_impl(IGPUImageView::SCreationParams&& params) override
     {
-        return nullptr;
+        VkImageViewCreateInfo vk_createInfo = { VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO };
+        vk_createInfo.pNext = nullptr; // Each pNext member of any structure (including this one) in the pNext chain must be either NULL or a pointer to a valid instance of VkImageViewASTCDecodeModeEXT, VkImageViewUsageCreateInfo, VkSamplerYcbcrConversionInfo, VkVideoProfileKHR, or VkVideoProfilesKHR
+        vk_createInfo.flags = static_cast<VkImageViewCreateFlags>(params.flags);
+
+        if (params.image->getAPIType() != EAT_VULKAN)
+            return nullptr;
+
+        VkImage vk_image = static_cast<const CVulkanImage*>(params.image.get())->getInternalObject();
+        vk_createInfo.image = vk_image;
+        vk_createInfo.viewType = static_cast<VkImageViewType>(params.viewType);
+        vk_createInfo.format = getVkFormatFromFormat(params.format);
+        vk_createInfo.components.r = static_cast<VkComponentSwizzle>(params.components.r);
+        vk_createInfo.components.g = static_cast<VkComponentSwizzle>(params.components.g);
+        vk_createInfo.components.b = static_cast<VkComponentSwizzle>(params.components.b);
+        vk_createInfo.components.a = static_cast<VkComponentSwizzle>(params.components.a);
+        vk_createInfo.subresourceRange.aspectMask = static_cast<VkImageAspectFlags>(params.subresourceRange.aspectMask);
+        vk_createInfo.subresourceRange.baseMipLevel = params.subresourceRange.baseMipLevel;
+        vk_createInfo.subresourceRange.levelCount = params.subresourceRange.levelCount;
+        vk_createInfo.subresourceRange.baseArrayLayer = params.subresourceRange.baseArrayLayer;
+        vk_createInfo.subresourceRange.layerCount = params.subresourceRange.layerCount;
+
+        VkImageView vk_imageView;
+        if (vkCreateImageView(m_vkdev, &vk_createInfo, nullptr, &vk_imageView) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<CVulkanImageView>(core::smart_refctd_ptr<CVKLogicalDevice>(this),
+                std::move(params), vk_imageView);
+        }
+        else
+        {
+            return nullptr;
+        }
     }
 
     core::smart_refctd_ptr<IGPUDescriptorSet> createGPUDescriptorSet_impl(IDescriptorPool* pool, core::smart_refctd_ptr<const IGPUDescriptorSetLayout>&& layout) override
     {
-        return nullptr;
+        if (pool->getAPIType() != EAT_VULKAN)
+            return nullptr;
+
+        const CVulkanDescriptorPool* vulkanDescriptorPool = static_cast<const CVulkanDescriptorPool*>(pool);
+
+        VkDescriptorSetAllocateInfo vk_allocateInfo = { VK_STRUCTURE_TYPE_DESCRIPTOR_SET_ALLOCATE_INFO };
+        vk_allocateInfo.pNext = nullptr; // pNext must be NULL or a pointer to a valid instance of VkDescriptorSetVariableDescriptorCountAllocateInfo
```

Comment:
what is `VkDescriptorSetVariableDescriptorCountAllocateInfo` for?

Reply:
I believe It's the case where descriptorIndexing device extension is enabled and also ``physDevDescriptorIndexingFeatures.descriptorBindingVariableDescriptorCount`` is equal to `VK_TRUE`  
meaning you'd enable variable sized descriptor arrays for shaders. (must be the last binding in the set)

So you don't know your the length of your variable-sized descriptor up to the point you actually want to allocate it's descriptorSet from the descriptorPool. This is what this struct is for; and It's basically asking for the "Count" of the "VariableCount" descriptors.

> descriptorBindingVariableDescriptorCount indicates whether the implementation supports descriptor sets with a variable-sized last binding. If this feature is not enabled, VK_DESCRIPTOR_BINDING_VARIABLE_DESCRIPTOR_COUNT_BIT must not be used.

In order to expose this``IGPUDescriptorSetLayout::SBinding`` should have an enum (a flag) related to this Vulkan Enum: 
```
typedef enum VkDescriptorBindingFlagBits {
    VK_DESCRIPTOR_BINDING_UPDATE_AFTER_BIND_BIT = 0x00000001,
    VK_DESCRIPTOR_BINDING_UPDATE_UNUSED_WHILE_PENDING_BIT = 0x00000002,
    VK_DESCRIPTOR_BINDING_PARTIALLY_BOUND_BIT = 0x00000004,
    VK_DESCRIPTOR_BINDING_VARIABLE_DESCRIPTOR_COUNT_BIT = 0x00000008,
.
.
.
} VkDescriptorBindingFlagBits;
```

In order for it to work you should also give the binding the correct flag when creating the `DescriptorSetLayout`. 
``VkDescriptorSetLayoutBindingFlagsCreateInfoEXT`` in the pNext chain of ``VkDescriptorSetLayoutCreateInfo``

Also we can do some validation ourselves here: for example check if the binding with the flag`VARIABLE_DESCRIPTOR_COUNT_BIT` is the last binding number in the set.

as mentioned here in the commetns https://github.com/Devsh-Graphics-Programming/Nabla/blob/6634885ebe66f4e9ba1657631f0902e0773ff7ba/src/nbl/video/CVKLogicalDevice.h#L1003

Reply:
@Erfan-Ahmadi turn this into an "enchancement" issue "Support Descriptor Indexing in Vulkan"

Reply:
Done  https://github.com/Devsh-Graphics-Programming/Nabla/issues/188